src/ValueObject/Valid/Validated.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Membrane\OpenAPIReader\ValueObject\Valid;
+
+abstract class Validated implements HasIdentifier, HasWarnings
+{
+    private Warnings $warnings;
+
+    public function __construct(
+        private readonly Identifier $identifier,
+    ) {
+    }
+
+    public function getIdentifier(): Identifier
+    {
+        return $this->identifier;
+    }
+
+    protected function appendedIdentifier(
+        string $primaryId,
+        string $secondaryId = ''
+    ): Identifier {
+        return $this->identifier->append($primaryId, $secondaryId);
+    }
+
+
+    public function hasWarnings(): bool
+    {
+        return isset($this->warnings);
+    }
+
+    public function getWarnings(): Warnings
+    {
+        if (!isset($this->warnings)) {
```

Comment:
this changes the state of the object; if you call

hasWarnings() -> false
getWarnings() -> empty Warnings object
hasWarnings() -> true

Change to either return the empty object without setting it or add an isEmpty method to the warnings object, use that for the hasWarningsCheck and initialise an empty object in the constructor

Reply:
Good spot, fixed.